R/utils.R

```diff
@@ -17,24 +17,47 @@ formatSolveBioResponse <- function (res, raw = FALSE) {
         res$'_url' = url
         return(res)
     }
+
 }
 
-formatSolveBioQueryResponse <- function (res, raw = FALSE, row.names = NULL) {
+formatSolveBioQueryResponse <- function (id, res, raw = FALSE, row.names = NULL) {
     # res will be the output of formatSolveBioResponse
     if (!raw & is.data.frame(res$results)) {
-        # Flatten the data frame
+        # Append column "_id" if it's not already there
+        if (!"_id" %in% row.names) {
+            row.names <- c(row.names, "_id")
+        }
+        # Remove columns from row.names that are not in the results query
+        diff <- setdiff(row.names, colnames(res$results))
+        row.names <- row.names [! row.names %in% diff]
```

Comment:
Looks like you are really getting the hang of R!

Can this be changed to just do something like `row.names[colnames(res$results)]` ? just assuming based on what the code looks like :)

Reply:
Done.